src/xdsl/printer.py

```diff
@@ -115,15 +115,19 @@ def _print_message(self,
     T = TypeVar('T')
 
     def print_list(self,
-                   elems: FrozenList[T] | List[T],
+                   elems: Iterable[T],
                    print_fn: Callable[[T], None],
                    delimiter: str = ", ") -> None:
-        if len(elems) == 0:
+        it = iter(elems)
```

Comment:
I don't actually see the difference in behavior between those two versions. Is it just that iterable has a different interface, so you changed it?

Reply:
Yeah, it doesn't need to be a list, can be any iterable, including generators, which is what I pass in the new function